builder/tart/step_run.go

```diff
@@ -100,28 +101,33 @@ func (u uiWriter) Write(p []byte) (n int, err error) {
 func (s *stepRun) Cleanup(state multistep.StateBag) {
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packersdk.Ui)
-
-	communicator := state.Get("communicator")
-	if communicator != nil {
+	cmd := state.Get("tart-cmd").(*exec.Cmd)
+	if cmd == nil {
+		return // Nothing to shut down
+	}
+	if cmd.ProcessState == nil {
```

Comment:
1. I think if we've got the `*exec.Cmd` from `"tart-cmd"` and it's non-`nil`, the `cmd.ProcessState == nil` will always evaluate to `false`. I'm missing something here?

2. Perhaps we should just skip this check altogether?

Reply:
ProcessState is set only after the process finishes, as documented here https://pkg.go.dev/os/exec#Cmd

Reply:
It's won't be magically set on its own, only after `Run()` or `Wait()`, which we won't call at this point in code.

We probably don't need this check at all, and the graceful shutdown process should be instead driven by the `state.Get("communicator")` being non-`nil`.

Reply:
Ah, I assumed it would be set also if the process finished on its own. The goal was to detect whether the process was already stopped/cleaned up.

Reply:
> We probably don't need this check at all, and the graceful shutdown process should be instead driven by the `state.Get("communicator")` being non-`nil`.

Removed the check. I assume you meant " driven by the `state.Get("communicator")` being *`nil`*".